src/endpoint.jl

```diff
@@ -1,3 +1,6 @@
+# isnothing doesn't exist in Julia 1.0
+isunbounded(a) = a === nothing
```

Comment:
calling this `isunbounded` rather than `isbounded` means the code has lots of double-negatives `!isunbounded` ...i wonder if flipping the condition would make the long chains of `&&` and `||` a little easier to read
```suggestion
isbounded(a) = a !== nothing
```